src/dsp_tools/utils/connection_live.py

```diff
@@ -395,5 +310,57 @@ def delete(
             headers=headers,
             timeout=timeout,
         )
-        check_for_api_error(response)
         return cast(dict[str, Any], response.json())
+
+    def _try_network_action(self, action: Callable[..., Any]) -> Any:
```

Comment:
Also, the return type could be `Response`, so the type annotations on the caller side would not be necessary

Reply:
But then we have to restrict the `action`, and make sure that it really always is a `requests.xyz()` callable. Otherwise we cannot safely claim that the return value is a `requests.Response`. Or do you think we should still do it?

Reply:
so you mean

```python
def _try_network_action(self, action: Callable[..., Response]) -> Response:
    ...
```

I don't see why not; and as it is a private method, we don't have to expect any other callers than the ones we know about in this file.

Reply:
actually, we don't need the `...` either, because the params have already been applied, so we know it's a callable that doesn't take any parameters, right?

Reply:
https://docs.python.org/3/library/typing.html#annotating-callable-objects

Reply:
Indeed, I can use `Callable[[], requests.Response]`, thanks for pointing that out :-)